src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -2586,40 +2619,50 @@ namespace HVACUnitarySystem {
+                ShowWarningError(RoutineName + ": " + CompType + " = " + CompName);
+                ShowContinueError("Number of cooling speeds does not match coil object.");
+                ShowFatalError("Cooling coil = " + VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).VarSpeedCoilType + ": " + VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).Name);
+            }
             DXCoolCap = GetCoilCapacityVariableSpeed(
                 cAllCoilTypes(UnitarySystem(UnitarySysNum).CoolingCoilType_Num), UnitarySystem(UnitarySysNum).CoolingCoilName, ErrFound);
             EqSizing.DesCoolingLoad = DXCoolCap;
 
-            for (Iter = 1; Iter <= UnitarySystem(UnitarySysNum).NumOfSpeedCooling; ++Iter) {
+            for (Iter = NumSpeeds; Iter >= 1;
+                 --Iter) { // use reverse order since we divide by CoolVolumeFlowRate(max)
                 UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) =
                     VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).MSRatedAirVolFlowRate(Iter);
                 UnitarySystem(UnitarySysNum).CoolMassFlowRate(Iter) = UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) * StdRhoAir;
-                // it seems the ratio should reference the actual flow rates, not the fan flow ???
-                if (UnitarySystem(UnitarySysNum).DesignFanVolFlowRate > 0.0 && UnitarySystem(UnitarySysNum).FanExists) {
-                    //             UnitarySystem(UnitarySysNum)%CoolVolumeFlowRate(UnitarySystem(UnitarySysNum)%NumOfSpeedCooling)
-                    UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(Iter) =
-                        UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) / UnitarySystem(UnitarySysNum).DesignFanVolFlowRate;
-                } else {
-                    UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(Iter) =
-                        UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) /
-                        UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(UnitarySystem(UnitarySysNum).NumOfSpeedCooling);
-                }
+                UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(Iter) =
+                    UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) /
+                    UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds);
             }
 
-            UnitarySystem(UnitarySysNum).IdleVolumeAirRate = UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(1);
-            UnitarySystem(UnitarySysNum).IdleMassFlowRate = UnitarySystem(UnitarySysNum).CoolMassFlowRate(1);
-            UnitarySystem(UnitarySysNum).IdleSpeedRatio = UnitarySystem(UnitarySysNum).MSCoolingSpeedRatio(1);
+            if (MSHPIndex > 0) {
+                UnitarySystem(UnitarySysNum).IdleVolumeAirRate =
+                    UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
```

Comment:
The current code produces the expected result for the defect file.
Still waiting to see all occurrences of `UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds)` replaced with `UnitarySystem(UnitarySysNum).MaxCoolAirVolFlow` (likewise for heating).